buku

```diff
@@ -599,21 +623,45 @@ class BukuDb:
         rows = self._fetch(query + ' LIMIT 1', *args, lock=lock)
         return rows[0] if rows else None
 
-    def get_rec_all(self, *, lock: bool = True):
+    def _ordering(self, fields=['+id'], for_db=True) -> List[Tuple[str, bool]]:
+        """Converts field list to ordering parameters (for DB query or entity list sorting).
+        Fields are listed in priority order, with '+'/'-' prefix signifying ASC/DESC; assuming ASC if not specified.
+        Other than names from DB, you can pass those from JSON export."""
+        names = {'index': 'id', 'uri': 'url', 'description': 'desc', **({'title': 'metadata'} if for_db else {'metadata': 'title'})}
+        valid = list(names) + list(names.values()) + ['tags']
+        _fields = [(re.sub(r'^[+-]', '', s), not s.startswith('-')) for s in (fields or [])]
+        _fields = [(names.get(field, field), direction) for field, direction in _fields if field in valid]
+        return _fields or [('id', True)]
```

Comment:
Parsing and validating an ordering description, converting it into a simple format for internal use.